snap/local/runtime-helpers/bin/security-secretstore-setup-launch.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash -ex
+
+security-secretstore-setup -confdir "$SNAP_DATA/config/security-secretstore-setup/res" --vaultInterval=10
+
+# if redis pw file doesn't exist, handle initialization & read it
+if [ ! -f "$REDIS5_PASSWORD_PATHNAME" ]; then
```

Comment:
Sorry to be a bother about this, but I'm still rather confused, since AFAICT, the thing that will generate the password file is `security-secretstore-setup` which we literally call right before this check in this script, so the only situation that this condition would be false is if security-secretstore-setup is configured to not run generate a redis password? Is that a supported configuration for the snap to configure security-secretstore-setup to _not_ use a password for redis? That also won't be the default configuration since it seems to generate this password on initial install, so not only would the user need to configure `security-secretstore-setup` to not generate a password, but they would also need to _delete_ the existing password too. Seems like a lot of work and unlikely to ever be done by a user.

All that is to say, I think we could probably get away with not having this check.

Reply:
In a perfect world, security-secretstore-setup would redis would read it's password directly from vault, but this isn't supported. In a less perfect world, `security-secretstore-setup` would read the password from vault and write it out to disk, but this isn't supported either. In the docker deployment, `security-secretstore-read` is called from the [entrypoint.sh](https://github.com/edgexfoundry/edgex-go/blob/master/cmd/security-secretstore-setup/entrypoint.sh#L54), which is why this wrapper script exists in the first place. So, when this script is first called, the password file will **not** exist, hence the check...